### PR TITLE
Allow events without "event" as the parent element in default mapping

### PR DESCRIPTION
**Pull request by minudika**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java [hunk 1]
```diff
@@ -63,7 +64,9 @@
         description = "JSON to Event input mapper. Transports which accepts JSON messages can utilize this extension"
                 + "to convert the incoming JSON message to Siddhi event. Users can either send a pre-defined JSON "
                 + "format where event conversion will happen without any configs or can use json path to map from a "
-                + "custom JSON message.",
+                + "custom JSON message.\n"
+                + "In default mapping either the json string of the event should be enclosed by the key \"event\""
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java]: Having \"event\" is optional right, but it's not informed.

File: component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java [hunk 2]
```diff
@@ -306,24 +320,21 @@ private Event convertToSingleEventForDefaultMapping(Object eventObject) throws I
...
-                } else {
-                    log.error("Default json message " + eventObject
-                            + " contains an invalid event identifier. Required \"event\", " +
-                            "but found \"" + parser.getText() + "\". Hence dropping the message.");
-                    return null;
+                    jsonToken = parser.nextToken();
                 }
-            } else if (JsonToken.FIELD_NAME.equals(jsonToken)) {
+            }
+            if (JsonToken.FIELD_NAME.equals(jsonToken)) {
```
Comment (suhothayan) [inline component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java:328]: What happen's if the event is like 
{\"testEvent\":{\"symbol\":\"WSO2\",\"price\":56.6,\"volume\":99}} 

We should not allow it to be mapped via default mapping

Comment (minudika) [reply component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java:328]: If fail.on.missing.attribute option is true, that event will be dropped.
If fail.on.missing.attribute option is false, the null values will be assigned to the event attributes.

Should we drop such events in the first place without considering fail.on.missing.attribute option?

Comment (minudika) [reply component/src/main/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapper.java:328]: fixed with bc34a00c149c8c8c81f2f35d3bf1c643d09dbbbd

**Review by suhothayan — APPROVED** [review]

**Pull request merged**